src/mlpack/methods/sparse_autoencoder/sparse_autoencoder.hpp

```diff
@@ -62,9 +62,7 @@ namespace nn {
  * @tparam OptimizerType The optimizer to use; by default this is L-BFGS.  Any
  *     mlpack optimizer can be used here.
  */
-template<
-  template<typename> class OptimizerType = mlpack::optimization::L_BFGS
->
+template<typename OptimizerType = mlpack::optimization::L_BFGS>
```

Comment:
Same here, I think only `Train()` and the constructors that train the model need to have `OptimizerType` as a template parameter.